lib/src/native/realm_core.dart

```diff
@@ -148,40 +147,231 @@ class _RealmCore {
+  }
+
+  SubscriptionHandle subscriptionAt(SubscriptionSet subscriptions, int index) {
+    return SubscriptionHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_subscription_at(
+          subscriptions.handle._pointer,
+          index,
+        )));
+  }
+
+  SubscriptionHandle? findSubscriptionByName(SubscriptionSet subscriptions, String name) {
+    return using((arena) {
+      return _realmLib
+          .realm_sync_find_subscription_by_name(
+            subscriptions.handle._pointer,
+            name.toUtf8Ptr(arena),
+          )
+          .convert(SubscriptionHandle._);
+    });
+  }
+
+  SubscriptionHandle? findSubscriptionByQuery(SubscriptionSet subscriptions, RealmResults query) {
+    return _realmLib
+        .realm_sync_find_subscription_by_results(
+          subscriptions.handle._pointer,
+          query.handle._pointer,
+        )
+        .convert(SubscriptionHandle._);
+  }
+
+  static void _stateChangeCallback(Pointer<Void> userdata, int state) {
+    final completer = userdata.toObject<Completer<SubscriptionSetState>>(isPersistent: true);
+    if (completer == null) {
+      return;
+    }
+    completer.complete(SubscriptionSetState.values[state]);
+  }
+
+  Future<SubscriptionSetState> waitForSubscriptionSetStateChange(SubscriptionSet subscriptions, SubscriptionSetState notifyWhen) {
+    final completer = Completer<SubscriptionSetState>();
+    _realmLib.realm_dart_sync_on_subscription_set_state_change_async(
+      subscriptions.handle._pointer,
+      notifyWhen.index,
+      Pointer.fromFunction(_stateChangeCallback),
+      completer.toPersistentHandle(),
+      _deletePersistentHandleFuncPtr,
+      subscriptions.realm.scheduler.handle._pointer,
+    );
+    return completer.future;
+  }
+
+  int subscriptionSetGetVersion(SubscriptionSet subscriptions) {
+    return _realmLib.realm_sync_subscription_set_version(subscriptions.handle._pointer);
+  }
+
+  SubscriptionSetState subscriptionSetGetState(SubscriptionSet subscriptions) {
+    return SubscriptionSetState.values[_realmLib.realm_sync_subscription_set_state(subscriptions.handle._pointer)];
+  }
+
+  MutableSubscriptionSetHandle subscriptionSetMakeMutable(SubscriptionSet subscriptions) {
+    return MutableSubscriptionSetHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_make_subscription_set_mutable(subscriptions.handle._pointer)));
+  }
+
+  SubscriptionSetHandle subscriptionSetCommit(MutableSubscriptionSet subscriptions) {
+    return SubscriptionSetHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_subscription_set_commit(subscriptions.handle._mutablePointer)));
+  }
+
+  SubscriptionHandle insertOrAssignSubscription(MutableSubscriptionSet subscriptions, RealmResults query, String? name, bool update) {
+    if (!update) {
+      if (name != null && findSubscriptionByName(subscriptions, name) != null) {
+        throw RealmException('Duplicate subscription $query${name != null ? ' with name: $name' : ''}');
```

Comment:
We did for a while using `realm_query_get_description`. That is no longer possible, since we are dropping the query handle. I will simplify this message.